kayak_render_macros/src/lib.rs

```diff
@@ -145,8 +150,23 @@ pub fn dyn_partial_eq(_: TokenStream, input: TokenStream) -> TokenStream {
 /// ```
 #[proc_macro]
 pub fn use_state(initial_state: TokenStream) -> TokenStream {
+    let state: proc_macro2::TokenStream = use_raw_state(initial_state).into();
+    let result = quote! {{
+        let (state, set_state, ..) = #state;
+        (state, set_state)
+    }};
+    TokenStream::from(result)
+}
+
+/// This macro works exactly like its [use_state] counterpart, except that it also returns
+/// the raw state Binding as the third field in the tuple.
+///
+/// returns: (state, set_state, state_binding)
+#[proc_macro]
+pub fn use_raw_state(initial_state: TokenStream) -> TokenStream {
```

Comment:
I'm debating on this. I added `use_raw_state` so we could access the actual state `Binding`, for use in the `use_effect` hook. But we could also just add that to the existing `use_state` hook, which could be cleaner.

However, doing this would require we create states like this, when we don't care about the raw `Binding`:

```rust
// We now have to add the ", .." part
let (state, set_state, ..) = use_state!(0usize);
```

I don't think it's the worst, but I'm not sure. Any thoughts?

Reply:
I'm wondering if it might be better to destructure the binding somehow. 🤔 something like:
```rust
let binding = context.create_state(0usize);
let (state, set_state) = binding.get_state();
```

I think I prefer this though:
```rust
let (state, set_state, ..) = use_state!(0usize);
```

Reply:
Ooh that would nice. We could maybe do that with a newtype around `Binding`, then impl `Deref` and stuff to access the raw `Binding`?

Then we wouldn't even really need the `use_state` macro which might be nice.

But yeah for now, I think I prefer 86ing the `use_raw_state`. I'll move the logic over to `use_state`.

Reply:
yeah I want to move flo_binding directly into this project since it doesn't seem like its updated anymore.

Reply:
Yeah that would be really nice. There are some things that could be more specialized for Kayak's purposes.